druid-shell/src/backend/wayland/surfaces/layershell.rs

```diff
@@ -369,6 +371,14 @@ impl Handle for Surface {
         return self.inner.wl_surface.borrow().invalidate_rect(rect);
     }
 
+    fn open_file(&self, options: FileDialogOptions) -> Option<FileDialogToken> {
```

Comment:
minor question: do we need to add these to every surface or can we just use the already implemented data() method to access the implementations directly vs expanding the api surface of handle.

Reply:
I was kinda wondering that too, but it seemed better to just follow what all the other methods were doing... I'd be happy to change it, though

Reply:
the PR is fine strictly speaking as is; since you're basically following the pattern every backend is doing.... its just a nit i have with druids internals. these methods shouldn't need to be nested as deeply into the backends as they are. a backend really should not care about opening file dialogs. i've covered the topic in detail in other issues.

looks like you're only using the idle handle. which is already exposed, and long term the wayland id. I'd try to keep the file stuff as lightly coupled as possible personally. but rather see functionality implemented than worry about architectural issues that impact every backend currently =)